### PR TITLE
[COM-186] Implement loading state for button component

### PR DESCRIPTION
**Pull request by Angele91**
## Please delete checklist items and sections that are not relevant

**Checklist**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Please check if the PR fulfills these requirements**

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Storybook or docs have been added / updated (for bug fixes / features)

**What is the ticket / issue associated with this change?** (Link to the ticket or issue)

https://impulsumvc.atlassian.net/browse/COM-186

**Storybook Link** (Link to a local storybook url to review new functionality)

http://localhost:8080/?path=/story/components-primitives-button--all

### CODE REVIEW

**Review by tkanzakic — CHANGES_REQUESTED** [review]
The size of the spinner does not match the design in figma

File: packages/components/src/components/primitives/Button/Button.tsx [hunk 1]
```diff
@@ -1,5 +1,20 @@
 import React, { FC } from 'react';
-import { Button as ChakraButton } from '@chakra-ui/react';
+import { Button as ChakraButton, useMultiStyleConfig } from '@chakra-ui/react';
 import { ButtonProps } from './types';
+import { Spinner } from '..';
+import { SpinnerProps } from '../Spinner/types';
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/Button/Button.tsx:5]: please re-export the `SpinnerProps` from `components/primitives/Spinner` instead, we missed that for some components

File: packages/components/src/components/primitives/Button/Button.tsx [hunk 2]
```diff
@@ -1,5 +1,20 @@
...
+export const Button: FC<ButtonProps> = (props) => {
+  const { loading } = useMultiStyleConfig('Button', {
+    variant: props.variant,
+    size: props.size,
+    isLoading: props.isLoading,
+  }) as { loading: SpinnerProps };
+
+  return (
+    <ChakraButton
+      spinner={<Spinner data-testid="cmpsr.button-spinner" {...loading} />}
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/Button/Button.tsx:16]: nit: `cmpsr.button.spinner`

File: packages/components/src/theme/ComposerTheme/Components/Button.ts [hunk 3]
```diff
@@ -35,11 +52,27 @@ const generateAltButton = (color: string) => {
...
+    borderColor: `${color}-default`,
+    borderBottomColor: transparentize(`${color}-default`, 0.3),
+    borderLeftColor: transparentize(`${color}-default`, 0.3),
+  };
+
+  if (color === 'secondary') {
+    loadingStyles = {
+      borderColor: `primary-default`,
+      borderBottomColor: transparentize(`primary-default`, 0.3),
+      borderLeftColor: transparentize(`primary-default`, 0.3),
```
Comment (tkanzakic) [inline packages/components/src/theme/ComposerTheme/Components/Button.ts:66]: no need to interpolate these strings

File: packages/components/src/components/primitives/Button/Button.stories.tsx [hunk 4]
```diff
@@ -12,6 +12,9 @@ export default {
       options: buttonVariants,
       control: { type: 'select' },
     },
+    isLoading: {
+      control: { type: 'boolean' },
+    },
```
[the remark this answers is not in the document]
Comment (Angele91) [reply packages/components/src/components/primitives/Button/Button.stories.tsx]: done!

File: packages/components/src/components/primitives/Button/Button.tsx (hunk 1 above)
Comment (Angele91) [reply packages/components/src/components/primitives/Button/Button.tsx]: done!

File: packages/components/src/components/primitives/Button/Button.tsx (hunk 2 above)
Comment (Angele91) [reply packages/components/src/components/primitives/Button/Button.tsx]: done!

File: packages/components/src/theme/ComposerTheme/Components/Button.ts (hunk 3 above)
Comment (Angele91) [reply packages/components/src/theme/ComposerTheme/Components/Button.ts]: done!

**Review by tkanzakic — APPROVED** [review]

**Pull request merged**